cmd/selfupdate/selfupdate.go

```diff
@@ -336,11 +336,30 @@ func makeRandomExeName(baseName, extension string) (string, error) {
 
 func downloadUpdate(ctx context.Context, beta bool, version, siteURL, newFile, packageFormat string) error {
 	osName := runtime.GOOS
-	arch := runtime.GOARCH
 	if osName == "darwin" {
 		osName = "osx"
 	}
-
+	arch := runtime.GOARCH
+	if arch == "arm" {
+		switch buildinfo.GetSupportedGOARM() {
+		case 7:
+			// This system can run any binaries built with GOARCH=arm, including GOARM=7.
+			// Pick the ARMv7 variant of rclone, published with suffix "arm-v7".
+			// Note that we don't know if that is currently running, it could be an
+			// ARMv6 variant in which case we will migrate to the better variant.
+			arch = "arm-v7"
+		case 6:
+			// This system can run binaries built with GOARCH=arm and GOARM=6 or lower.
+			// Pick the ARMv6 variant of rclone, published with suffix "arm".
+			// Note that we don't know if that is currently running, it could be an
+			// ARMv5 variant in which case we will migrate to the better variant.
+			arch = "arm"
+		case 5:
+			// This system can only run binaries built with GOARCH=arm and GOARM=5.
+			// Rclone does not currently publish such variants, i.e. selfupdate is not possible.
```

Comment:
We don't? Didn't I see them just a commit or two ago?

Reply:
Ah yes, I think this now just needs to be squashed together with the next commit ("consider arm version (improved version after ARM v5 vs v6 fix)".